tket/tests/test_AASRoute.cpp

```diff
@@ -0,0 +1,429 @@
+
+    std::map<UnitID, UnitID> rename_map = {
+        {qubits[0], nodes[0]}, {qubits[1], nodes[1]},  {qubits[2], nodes[2]},
+        {qubits[3], nodes[3]}, {qubits[4], nodes[4]},  {qubits[5], nodes[5]},
+        {qubits[6], nodes[6]}, {qubits[7], nodes[7]},  {qubits[8], nodes[8]},
+        {qubits[9], nodes[9]}, {qubits[10], nodes[10]}};
+
+    circ.rename_units(rename_map);
+
+    std::shared_ptr<MappingFrontier> mf =
+        std::make_shared<MappingFrontier>(circ);
+
+    AASRouteRoutingMethod aasrm(aas::CNotSynthType::Rec, 1);
+
+    REQUIRE(!aasrm.check_method(mf, shared_arc));
+  }
+  GIVEN("test check_method in AASRouteRoutingMethod II") {
+    Circuit circ(11);
+    std::vector<Qubit> qubits = circ.all_qubits();
+
+    std::map<UnitID, UnitID> rename_map = {
+        {qubits[0], nodes[0]}, {qubits[1], nodes[1]},  {qubits[2], nodes[2]},
+        {qubits[3], nodes[3]}, {qubits[4], nodes[4]},  {qubits[5], nodes[5]},
+        {qubits[6], nodes[6]}, {qubits[7], nodes[7]},  {qubits[8], nodes[8]},
+        {qubits[9], nodes[9]}, {qubits[10], nodes[10]}};
+
+    Circuit ppb_circ(11);
+
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[0], qubits[4]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[6], qubits[7]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[1], qubits[10]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[8], qubits[5]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[3], qubits[9]});
+
+    PhasePolyBox ppbox(ppb_circ);
+    circ.add_box(ppbox, qubits);
+
+    circ.rename_units(rename_map);
+
+    std::shared_ptr<MappingFrontier> mf =
+        std::make_shared<MappingFrontier>(circ);
+
+    AASRouteRoutingMethod aasrm(aas::CNotSynthType::Rec, 1);
+
+    REQUIRE(aasrm.check_method(mf, shared_arc));
+  }
+  GIVEN("test check_method in AASRouteRoutingMethod III") {
+    Circuit circ(11);
+    std::vector<Qubit> qubits = circ.all_qubits();
+
+    std::map<UnitID, UnitID> rename_map = {
+        {qubits[0], nodes[0]}, {qubits[1], nodes[1]},  {qubits[2], nodes[2]},
+        {qubits[3], nodes[3]}, {qubits[4], nodes[4]},  {qubits[5], nodes[5]},
+        {qubits[6], nodes[6]}, {qubits[7], nodes[7]},  {qubits[8], nodes[8]},
+        {qubits[9], nodes[9]}, {qubits[10], nodes[10]}};
+
+    Circuit ppb_circ(11);
+
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[0], qubits[4]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[6], qubits[7]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[1], qubits[10]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[8], qubits[5]});
+    ppb_circ.add_op<UnitID>(OpType::CX, {qubits[3], qubits[9]});
+
+    PhasePolyBox ppbox(ppb_circ);
+    circ.add_box(ppbox, qubits);
+
+    std::shared_ptr<MappingFrontier> mf =
+        std::make_shared<MappingFrontier>(circ);
+
+    AASRouteRoutingMethod aasrm(aas::CNotSynthType::Rec, 1);
+
+    REQUIRE(!aasrm.check_method(mf, shared_arc));
+  }
+  GIVEN("test routing_method in AASRouteRoutingMethod I") {
```

Comment:
Similarly to other comment, please can you remove I, II and III etc and replace them with something more descriptive?